### ISSUE TITLE
[Bug] Null reference on validator even when manually setting Validator="new Validator()"

### ISSUE THREAD

**Issue by ScottKane** [Blazored/FluentValidation#152]
I'm getting a null reference on the validator for some strange reason, I have another form that uses the exact same method of validating. The validator for this form and the other are in the exact same place. The other validator works fine, but this one throws a null reference:

```
Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at KitchensPlus.Client.Components.Dialogs.AddEditCustomer.get_Validated() in D:\Work\KitchensPlus\Client\Components\Dialogs\AddEditCustomer.razor:line 111
```

```c#
<EditForm Model="@Model" OnValidSubmit="SubmitAsync">
<FluentValidationValidator @ref="@_validator" />

@code {
    private FluentValidationValidator _validator;
    private bool Validated => _validator.Validate(options => { options.IncludeAllRuleSets(); });
}
```

I've even tried setting the validator manually:
```c#
<FluentValidationValidator Validator="new AddEditCustomerCommandValidator()" @ref="@_validator" />
```

It still results in a null reference. Am I missing something here? It's driving me crazy.

**Screenshots**
![Screenshot 2022-07-25 101048](https://user-images.githubusercontent.com/17254944/180742148-70dadff1-6ceb-4382-b1d0-d1d213dc3fd0.png)
![Screenshot 2022-07-25 101020](https://user-images.githubusercontent.com/17254944/180742170-d215c9d1-b0d8-4c6d-8fa0-0d1142c9d922.png)
![Screenshot 2022-07-25 101132](https://user-images.githubusercontent.com/17254944/180742192-f8b9bfb7-938e-4d87-87ab-f8122b9b182e.png)
![Screenshot 2022-07-25 101156](https://user-images.githubusercontent.com/17254944/180742198-7d07e612-464a-4085-9f6e-234fa11da28e.png)
![Screenshot 2022-07-25 101217](https://user-images.githubusercontent.com/17254944/180742207-4d999756-2b1b-4cf8-8f83-a22791ecdaf0.png)

RegisterRequestValidator works fine, AddEditCustomerCommandValidator isn't working

**Hosting Model (is this issue happening with a certain hosting model?):**
Blazor WebAssembly

.NET 6.0.7
Latest preview version of Blazored.FluentValidation (tried non preview versions too)

**Comment by ScottKane** [comment]
I've also tried moving the nested object validation into separate validators to no avail.

![Screenshot 2022-07-25 104214](https://user-images.githubusercontent.com/17254944/180747953-aecaa602-b9fe-4236-a154-fea64a6a41ff.png)

**Comment by ScottKane** [comment]
See https://github.com/ScottKane/ValidatorNotWorking for simple repro

**Comment by ScottKane** [comment]
@chrissainty Sorry to tag you directly but this is a blocking issue for a testing phase and I'm being hounded, is there any chance you can take a quick look at the repro and see if you can figure out what's going on? Thanks in advance 😃

**Comment by chrissainty** [comment]
@ScottKane Im away till early next week but I will take a look when I get back.

**Comment by Sharaf-Mansour** [comment]
## TL;DR;
#### Do null Checking 
```cs
    private FluentValidationValidator? _validator;
    private bool? Validated => _validator?.Validate(options => { options.IncludeAllRuleSets(); });
```
#### and for the razor 
```cs
 <MudButton OnClick="SubmitAsync" DisableElevation Variant="Variant.Filled"
 Disabled='@(!Validated ?? false)' Color="Color.Secondary">@(Model.Id != 0 ? "Update" : "Save")</MudButton>
 ```


@ScottKane  I tried your repo and here is the issue I think we need to address.
Using @ref will trigger OnAfterRender.
So of course the _validator will be null because it is not pre-initialized 
```cs
    private FluentValidationValidator _validator;
```

You can understand more about how ref work in blazor here.
https://blazor-university.com/javascript-interop/calling-javascript-from-dotnet/passing-html-element-references/
and here.
https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0#capture-references-to-components
So as a workaround we just need to handle the Null with some logic
Something more like this
```cs
    private FluentValidationValidator? _validator;
    private bool? Validated => _validator?.Validate(options => { options.IncludeAllRuleSets(); });
```
and for the razor 
```cs
 <MudButton OnClick="SubmitAsync" DisableElevation Variant="Variant.Filled"
 Disabled='@(!Validated ?? false)' Color="Color.Secondary">@(Model.Id != 0 ? "Update" : "Save")</MudButton>
 ```
 To understand it more do this.
 ```cs
  @(_validator is null)
<button class="btn" @onclick="()=>StateHasChanged()">ref</button>
```
Before you press on the refresh button it will say true. meaning that _validator is indeed null, If you call **StateHasChanged** the _validator will be initialized and so it will say false...
And about why it is working fine with Register.razor file
Simply call ```@(Validated)```   And it will trigger the same error.

**Comment by ScottKane** [comment]
@Sharaf-Mansour Thanks for the reply, That makes sense, not sure I like that one works and one doesn't but it's good to know the fix is simple. I've moved to just initialising the validator manually with my own component while I was waiting so at least everything is working now.
